src/box/lua/xlog.c

```diff
@@ -95,7 +95,7 @@ lbox_xlog_parse_body_kv(struct lua_State *L, int type, const char **beg, const c
 	if (mp_typeof(**beg) != MP_UINT)
 		luaL_error(L, "Broken type of body key");
 	uint32_t v = mp_decode_uint(beg);
-	if (iproto_key_name(v)) {
+	if (iproto_type_is_dml(type) && iproto_key_name(v)) {
```

Comment:
`VY_INDEX_RUN_INFO`, `VY_INDEX_PAGE_INFO` and `VY_RUN_PAGE_INDEX` xrow types have their own namespaces for xrow->body keys. For example, there is `IPROTO_SYNC = 0x01`, but at the same time we have `VY_PAGE_INFO_OFFSET = 1`, `VY_RUN_INFO_MIN_LSN = 1` and `VY_PAGE_INDEX_INDEX = 1`. This change checks that request is DML (e.g REPLACE/INSERT/DELETE/UPSERT) first and only after that tries to use iproto_key_name().